src/java.base/share/classes/java/lang/reflect/ProxyGenerator.java

```diff
@@ -598,6 +599,14 @@ private void generateStaticInitializer() {
                 JL_CLASS_NOT_FOUND_EX);
 
         mv.visitLabel(L_startBlock);
+
+        // Put ClassLoader at local variable index 0, used by
+        // Class.forName(String, boolean, ClassLoader) calls
+        mv.visitLdcInsn(Type.getObjectType(dotToSlash(className)));
+        mv.visitMethodInsn(INVOKEVIRTUAL, JL_CLASS,
+                "getClassLoader", "()" + LJL_CLASSLOADER, false);
+        mv.visitVarInsn(ASTORE, 0);
```

Comment:
Shouldn’t this go before `mv.visitLabel(L_startBlock)`?

Reply:
Done.